client/app/scripts/components/nodes.js

```diff
@@ -94,9 +109,16 @@ class Nodes extends React.Component {
 
 function mapStateToProps(state) {
   return {
-    nodes: state.get('nodes'),
-    selectedNodeId: state.get('selectedNodeId'),
+    gridMode: state.get('gridMode'),
+    gridSortBy: state.get('gridSortBy'),
+    gridSortedDesc: state.get('gridSortedDesc'),
+    nodes: state.get('nodes').filter(node => !node.get('filtered')),
+    currentTopology: state.get('currentTopology'),
+    currentTopologyId: state.get('currentTopologyId'),
     topologyEmpty: isTopologyEmpty(state),
+    searchNodeMatches: state.getIn(['searchNodeMatches', state.get('currentTopologyId')]),
+    searchQuery: state.get('searchQuery'),
+    selectedNodeId: state.get('selectedNodeId')
   };
```

Comment:
The props needed only for gridview, should be put in the grid's `mapStateToProps`.